src/dg/weak_dg.cpp

```diff
@@ -2010,16 +1472,52 @@ void DGWeak<dim,nstate,real,MeshType>::assemble_boundary_term_derivatives(
 #endif
 
 
+
+template<int dim>
+dealii::Quadrature<dim> project_face_quadrature(
```

Comment:
Functions that don't need direct access to the internal class structure should not be a member of the class.

Kind of C.4 and C.5
https://www.modernescpp.com/index.php/c-core-guidelines-class-rules/

There is a bit of gray area between putting the function in an anonymous namespace versus in the namespace the class is in.

Reply:
That makes sense. Thanks for linking that style guide, it's quite helpful!

Arguable it may be better organization to re-order the functions in `weak_dg.cpp` such that all DGWeak member functions are grouped together, followed by non-member functions. But this would be a mostly aesthetic change and may be more work than it's worth, or disrupt proximity-based organization.